tests/t0114_gateway_subdomains_test.go

```diff
@@ -2,61 +2,562 @@ package tests
+			"request for example.com/ipfs/{InvalidCID} produces useful error before redirect",
+			"error message should include original CID (and it should be case-sensitive, as we can't assume everyone uses base32)",
+			Url("%s://%s/ipfs/QmInvalidCID", u.Scheme, u.Host),
+			Expect().
+				Body(Contains("invalid path \"/ipfs/QmInvalidCID\"")).
+				Response(),
+		))
+
+		with(testGatewayWithManyProtocols(t,
+			"request for example.com/ipfs/{CIDv0} produces redirect to {CIDv1}.ipfs.example.com",
+			"",
+			Url("%s://%s/ipfs/%s/", u.Scheme, u.Host, CIDv0),
+			Expect().
+				Status(301).
+				Headers(
+					Header("Location").Equals("%s://%s.ipfs.%s/", u.Scheme, CIDv0to1, u.Host),
+				).
+				Response(),
+		))
+
+		with(testGatewayWithManyProtocols(t,
+			"request for http://example.com/ipfs/{CID} with X-Forwarded-Proto: https produces redirect to HTTPS URL",
+			"Support X-Forwarded-Proto",
+			Request().
+				URL("%s://%s/ipfs/%s/", u.Scheme, u.Host, CIDv1).
+				Header("X-Forwarded-Proto", "https"),
+			Expect().
+				Status(301).
+				Headers(
+					Header("Location").Equals("https://%s.ipfs.%s/", CIDv1, u.Host),
+				).
+				Response(),
+		))
+
+		with(testGatewayWithManyProtocols(t,
+			"request for example.com/ipfs/?uri=ipfs%3A%2F%2F.. produces redirect to /ipfs/.. content path",
+			"Support ipfs:// in https://developer.mozilla.org/en-US/docs/Web/API/Navigator/registerProtocolHandler",
+			Request().
+				URL("%s://%s/ipfs/", u.Scheme, u.Host).
+				Query(
+					"uri", "ipfs://%s/wiki/Diego_Maradona.html", CIDMaradona,
+				),
+			Expect().
+				Status(301).
+				Headers(
+					Header("Location").Equals("/ipfs/%s/wiki/Diego_Maradona.html", CIDMaradona),
+				).
+				Response(),
+		))
+
+		// # example.com/ipns/<libp2p-key>
+		// TODO
+
+		// # example.com/ipns/<dnslink-fqdn>
+		// TODO
+
+		// # DNSLink on Public gateway with a single-level wildcard TLS cert
+		// # "Option C" from  https://github.com/ipfs/in-web-browsers/issues/169
+		// TODO
+
+		// # Support ipns:// in https://developer.mozilla.org/en-US/docs/Web/API/Navigator/registerProtocolHandler
+		// TODO
+
+		// # *.ipfs.example.com: subdomain requests made with custom FQDN in Host header
+		// request for {CID}.ipfs.example.com should return expected payload
+		// TODO(lidel): Note: skipped, this test exists above thanks to the for loop.
+
+		// request for {CID}.ipfs.example.com/ipfs/{CID} should return HTTP 404
+		// TODO(lidel): Note: skipped, this test exists above thanks to the for loop.
+
+		// # *.ipns.example.com
+		// # ============================================================================
+
+		// # <libp2p-key>.ipns.example.com
+
+		// # API on subdomain gateway example.com
+		// # ============================================================================
+
+		// # disable /api on example.com
+		// TODO(lidel): in this part of the test suite, we disable example.com then
+		// we test that the endpoints returns 404. This is more Kubo than conformance-related.
```

Comment:
The test is in: 
https://github.com/ipfs/kubo/blob/668d0b2fa4b43a1474ea0d1deab5295810a55f68/test/sharness/t0114-gateway-subdomains.sh#L537

My guess is that this should not live in the conformance test suite.